cloud/cloud-init.yml

```diff
@@ -10,13 +9,40 @@ packages:
  - gnupg
  - lsb-release
  - git
- 
+
 runcmd:
   - sudo mkdir -p /etc/apt/keyrings
-  - curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --dearmor -o /etc/apt/keyrings/docker.gpg
+  - curl -fsSL https://download.docker.com/linux/ubuntu/gpg | sudo gpg --import
+  - FINGERPRINT="9DC8 5822 9FC7 DD38 854A E2D8 8D81 803C 0EBF CD88"
+  - if [ "$(sudo gpg --verify-options show-notations --verify /etc/apt/keyrings/docker.gpg | grep "$FINGERPRINT" | wc -l)" -eq 0 ]; then echo "Docker keyring fingerprint does not match expected value." && exit 1; fi
   - echo "deb [arch=$(dpkg --print-architecture) signed-by=/etc/apt/keyrings/docker.gpg] https://download.docker.com/linux/ubuntu $(lsb_release -cs) stable" | sudo tee /etc/apt/sources.list.d/docker.list > /dev/null
   - sudo apt-get update
-  - sudo apt-get install -y docker-ce docker-ce-cli containerd.io docker-compose-plugin
-  - git clone https://github.com/WhatsApp/proxy.git $HOME/whatsapp-proxy
-  - docker compose -f $HOME/whatsapp-proxy/proxy/ops/docker-compose.yml up -d
-
+  - sudo apt-get install -y docker-ce=5:20.10.2~ce~3-0~ubuntu
+  - sudo useradd -r -s /sbin/nologin docker
+  - sudo usermod -aG docker $USER
+  - echo 'DOCKER_OPTS="-H unix:///var/run/docker.sock --userland-proxy-path /usr/lib/docker/docker-proxy-current -g /var/lib/docker --userland-proxy-user=docker --userland-proxy-group=docker"' | sudo tee -a /etc/default/docker
+  - sudo systemctl enable --now docker
+  - sudo iptables -I INPUT -p tcp --dport 2375 -j DROP
+  - sudo iptables -I INPUT -p tcp --dport 2376 -j DROP
+  - sudo iptables -I INPUT -p tcp --dport 2377 -j DROP
+  - sudo iptables -I INPUT -p tcp --dport 2378 -j DROP
```

Comment:
Why are we dropping these packets coming to these ports?

Reply:
You are right, it was supposed to be on my machine